### ISSUE TITLE
Can this method support any base model like a plug-in?

### ISSUE THREAD

**Issue by akk-123** [limuloo/MIGC#1]

**Comment by limuloo** [comment]
Thank you very much for your interest in our work. We have now shared MIGC trained on the SD1.4 base model, and it can work like a plug-in for any base model fine-tuned from SD1.x architecture (e.g., we can use MIGC in RV60B1, Cetus, Ghost base models without any training). If you want to use the base model fine-tuned from SDXL architecture, maybe We also need to train a version of MIGC on the SDXL base model, and we will implement this as soon as possible.

<img width="851" alt="image" src="https://github.com/limuloo/MIGC/assets/132473106/36e4d0f6-8553-4ee3-9e83-e3c1ae539d1b">

**Comment by akk-123** [comment]
Thank you! Looking forward to your sdxl model and the GUI (for easier testing). Additionally, it would be great if you could update the version of diffusers.

**Comment by akk-123** [comment]
@limuloo I try your method
```python 
# base_model use majicmixRealistic_v7
prompt_final = [['a woman and a man, poratrait, white background', 
                     'a young woman, portrait',
                     'a young man, portrait',
                     'red hair',
                     'green scarf',
                     'pink sweater',
                     'yellow hat',
                     'gray suit']]
    bboxes =[[[0.05, 0.25, 0.5, 0.98], 
              [0.5, 0.25, 0.9, 0.98],
              [0.1, 0.25, 0.5, 0.5],
              [0.05, 0.5, 0.45, 0.8],
              [0.05, 0.55, 0.5, 0.98],
              [0.55, 0.25, 0.75, 0.4],
              [0.5, 0.55, 0.9, 0.98]]]
```

below is reult, the overall effect is good, but it seems that some constraints did not work, such as the prompt is for a white background, the generated is  black.
![image](https://github.com/limuloo/MIGC/assets/98469560/bb97e4c7-f824-4cd5-a1a2-c9ebf4a9d23d)

**Comment by limuloo** [comment]
You can control the background as an instance, which will get better results. In fact, this is what we do during the training process. At the same time, you can also change the seed to get better results.

**Comment by limuloo** [comment]
> @limuloo I try your method
> 
> ```python
> # base_model use majicmixRealistic_v7
> prompt_final = [['a woman and a man, poratrait, white background', 
>                      'a young woman, portrait',
>                      'a young man, portrait',
>                      'red hair',
>                      'green scarf',
>                      'pink sweater',
>                      'yellow hat',
>                      'gray suit']]
>     bboxes =[[[0.05, 0.25, 0.5, 0.98], 
>               [0.5, 0.25, 0.9, 0.98],
>               [0.1, 0.25, 0.5, 0.5],
>               [0.05, 0.5, 0.45, 0.8],
>               [0.05, 0.55, 0.5, 0.98],
>               [0.55, 0.25, 0.75, 0.4],
>               [0.5, 0.55, 0.9, 0.98]]]
> ```
> 
> below is reult, the overall effect is good, but it seems that some constraints did not work, such as the prompt is for a white background, the generated is black. ![image](https://private-user-images.githubusercontent.com/98469560/310777831-bb97e4c7-f824-4cd5-a1a2-c9ebf4a9d23d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk4MDkxNzAsIm5iZiI6MTcwOTgwODg3MCwicGF0aCI6Ii85ODQ2OTU2MC8zMTA3Nzc4MzEtYmI5N2U0YzctZjgyNC00Y2Q1LWExYTItYzllYmY0YTlkMjNkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA3VDEwNTQzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFlODZmMDM4OGJiNDg4MDFkNzI5YzUwMjkzMWQzNDhhNDAzOWM0Y2M1NzlmMTE2ZDcyM2ZiN2U0NGU5MmE3ZWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ZZfcwdHVguTd3wo0dCUzXUmgu_zj3-gRBOVsbPhojjQ)

You can control the background as an instance, which will get better results. In fact, this is what we do during the training process. At the same time, you can also change the seed to get better results.
prompt_final = [['a woman and a man, poratrait, white background', 
                     'a young woman, portrait',
                     'a young man, portrait',
                     'red hair',
                     'green scarf',
                     'pink sweater',
                     'yellow hat',
                     'gray suit', 'white background']]
    bboxes =[[[0.05, 0.25, 0.5, 0.98], 
              [0.5, 0.25, 0.9, 0.98],
              [0.1, 0.25, 0.5, 0.5],
              [0.05, 0.5, 0.45, 0.8],
              [0.05, 0.55, 0.5, 0.98],
              [0.55, 0.25, 0.75, 0.4],
              [0.5, 0.55, 0.9, 0.98],
              [0, 0, 1.0, 1.0]]]
    negative_prompt = 'worst quality, low quality, bad anatomy, watermark, text, blurry'
    seed = 42
![anno_outputv2](https://github.com/limuloo/MIGC/assets/132473106/be28a77a-805f-4913-b4a3-6906ef3d2ecd)

**Comment by limuloo** [comment]
> @limuloo thanks for your reply what' more, I have a question for your code [here](https://github.com/limuloo/MIGC/blob/main/migc/migc_arch.py#L176), in the layout attention, you set
> 
> ```python
> guidance_mask=torch.cat([guidance_mask[:, 1:, ...], supplement_mask], dim=1))
> ```
> 
> why use guidance_mask[:, 1:, ...], this means not use first instance mask

Thank you for your careful reading. This should be an error caused when we were sorting out the code. The code you mentioned should be guidance_mask[:, :, ...]. We will fix this problem as soon as possible. Regarding the GUI code issues that you are concerned about, we will release the code this week so that everyone can use MIGC to create more conveniently.
![image](https://github.com/limuloo/MIGC/assets/132473106/84c7a584-71aa-4dde-9cba-cfda3e10d769)

**Comment by akk-123** [comment]
Get it!

**Comment by akk-123** [comment]
@limuloo hi, I have read the paper and the code, and I still have a few questions. It is a bit cumbersome to communicate here. Can you provide your contact information or email? I have a few questions to ask.

**Comment by limuloo** [comment]
If you have any questions, feel free to contact me via email zdw1999@zju.edu.cn

**Comment by akk-123** [comment]
I have sent an email

**Comment by limuloo** [comment]
在你这个case，就算三个相加起来以后，每个点i（每行）它能关注到的点j （每列）还是不变呀。

**Comment by limuloo** [comment]
左边框的点只能关注到左边框的点  右边框的点只能关注到右边框的点  中间背景只能关注到中间背景

**Comment by akk-123** [comment]
ok，我理解了
